sources/api/thar-be-updates/src/main.rs

```diff
@@ -0,0 +1,427 @@
+                    [ --log-level trace|debug|info|warn|error ]     Logging level (default info)",
+        program_name, DEFAULT_API_SOCKET,
+    );
+    process::exit(2);
+}
+
+/// Prints a more specific message before exiting through usage().
+fn usage_msg<S: AsRef<str>>(msg: S) -> ! {
+    eprintln!("{}\n", msg.as_ref());
+    usage();
+}
+
+/// Parses the command line arguments
+fn parse_args(args: std::env::Args) -> Args {
+    let mut subcommand = None;
+    let mut log_level = None;
+    let mut socket_path = None;
+
+    let mut iter = args.skip(1).peekable();
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "--log-level" => {
+                let log_level_str = iter
+                    .next()
+                    .unwrap_or_else(|| usage_msg("Did not give argument to --log-level"));
+                log_level = Some(LevelFilter::from_str(&log_level_str).unwrap_or_else(|_| {
+                    usage_msg(format!("Invalid log level '{}'", log_level_str))
+                }));
+            }
+
+            "--socket-path" => {
+                socket_path = Some(
+                    iter.next()
+                        .unwrap_or_else(|| usage_msg("Did not give argument to --socket-path")),
+                )
+            }
+            // Assume any arguments not prefixed with '-' is a subcommand
+            s if !s.starts_with('-') => {
+                if subcommand.is_some() {
+                    usage();
+                }
+                subcommand =
+                    Some(serde_plain::from_str::<UpdateCommand>(s).unwrap_or_else(|_| usage()));
+            }
+            _ => usage(),
+        }
+    }
+
+    Args {
+        subcommand: subcommand.unwrap_or_else(|| usage()),
+        log_level: log_level.unwrap_or_else(|| LevelFilter::Info),
+        socket_path: socket_path.unwrap_or_else(|| DEFAULT_API_SOCKET.to_string()),
+    }
+}
+
+// Some simple wrapper functions for locking
+// Once we fork, the parent process and child process are going to have duplicate file descriptors
+// that refer to the same lock. Once the parent returns and closes its copy of the lockfile fd,
+// the child will still hold the lock. The lock is only released when all copies of the file descriptor are closed.
+fn lock_exclusive(lockfile: &File) -> Result<()> {
+    lockfile.try_lock_exclusive().context(error::LockHeld {
+        path: UPDATE_LOCKFILE,
+    })?;
+    debug!("Obtained exclusive lock");
+    Ok(())
+}
+
+fn unlock(lockfile: &File) -> Result<()> {
+    lockfile.unlock().context(error::Unlock {
+        path: UPDATE_LOCKFILE,
+    })?;
+    debug!("Released lock");
+    Ok(())
+}
+
+/// Initializes the update status and creates the update status file
+fn initialize_update_status() -> Result<()> {
+    let mut new_status = UpdateStatus::new();
+    // Initialize active partition set information
+    new_status.update_active_partition_info()?;
+    write_update_status(&new_status)
+}
```

Comment:
I'd find it more readable if we were managing a "state" file rather than a "status" file. Especially in the case where we're doing it side by side with a `std::process::Output`. It's more of a durable view of the state of the update subsystem than a reflection of the last command to run.